cargo-dist/src/init.rs

```diff
@@ -69,6 +79,32 @@ pub fn do_init(cfg: &Config, args: &InitArgs) -> DistResult<()> {
 
     // Load in the root workspace toml to edit and write back
     let mut workspace_toml = config::load_cargo_toml(&root_workspace.manifest_path)?;
+    let mut original_workspace_toml = workspace_toml.clone();
+    let initted = has_metadata_table(root_workspace);
+
+    // If this is a Cargo.toml, offer to either write their config to
+    // a dist-workspace.toml, or migrate existing config there
+    let mut is_migrating = false;
+    let desired_workspace_kind = if root_workspace.kind == WorkspaceKind::Rust {
+        let prompt = if initted {
+            r#"Would you like to migrate your project from Cargo.toml to dist-workspace.toml?"#
```

Comment:
We'll probably either want this to be really verbose, or have a docs page to link to.